### PR TITLE
[redis-plus-plus] Add  c++17 feature

### LINKED ISSUE #14654: [redis-plus-plus] Build using c++17 standard
****

**Is your feature request related to a problem? Please describe.**
Redis-plus-plus is by default built using c++11 standard. When trying to use it in a Linux application built using C++17 I get link errors.

**Proposed solution**
Build redis-plus-plus using define flag REDIS_PLUS_PLUS_CXX_STANDARD=17.

```patch
diff --git a/ports/redis-plus-plus/portfile.cmake b/ports/redis-plus-plus/portfile.cmake
index fae747289..33f222c58 100644
--- a/ports/redis-plus-plus/portfile.cmake
+++ b/ports/redis-plus-plus/portfile.cmake
@@ -25,6 +25,7 @@ vcpkg_configure_cmake(
         -DREDIS_PLUS_PLUS_BUILD_STATIC=${REDIS_PLUS_PLUS_BUILD_STATIC}
         -DREDIS_PLUS_PLUS_BUILD_SHARED=${REDIS_PLUS_PLUS_BUILD_SHARED}
         -DREDIS_PLUS_PLUS_BUILD_TEST=OFF
+        -DREDIS_PLUS_PLUS_CXX_STANDARD=17
 )

 vcpkg_install_cmake()
@@ -34,4 +35,4 @@ vcpkg_copy_pdbs()
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)

 # Handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright )
\ No newline at end of file
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright )
```

**Additional context**
See CMakeLists.txt that contains REDIS_PLUS_PLUS_CXX_STANDARD flag: https://github.com/sewenew/redis-plus-plus/blob/a4b8a92e1ea6f1be8bb3ce6539b0a4ae6cf41b77/CMakeLists.txt#L9

### PR DESCRIPTION
**Pull request by JonLiu1993**
**Describe the pull request**

- What does your PR fix? Fixes #14654 
Redis-plus-plus is by default built using c++11 standard. When trying to use it in a Linux application built using C++17 I get link errors.

Build redis-plus-plus using define flag REDIS_PLUS_PLUS_CXX_STANDARD=17 can be solved

All featureas are tested successfully in the following triplets:

- x86-windows
- x64-windows
- x64-windows-static

### CODE REVIEW

File: ports/redis-plus-plus/portfile.cmake [hunk 1]
```diff
@@ -25,6 +25,7 @@ vcpkg_configure_cmake(
         -DREDIS_PLUS_PLUS_BUILD_STATIC=${REDIS_PLUS_PLUS_BUILD_STATIC}
         -DREDIS_PLUS_PLUS_BUILD_SHARED=${REDIS_PLUS_PLUS_BUILD_SHARED}
         -DREDIS_PLUS_PLUS_BUILD_TEST=OFF
+        -DREDIS_PLUS_PLUS_CXX_STANDARD=17
```
Comment (JackBoosY) [inline ports/redis-plus-plus/portfile.cmake:28]: Add a feature `cxx17` instead.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/redis-plus-plus/portfile.cmake [hunk 2]
```diff
@@ -9,6 +9,11 @@ vcpkg_from_github(
         fix-conversion.patch
 )
 
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    cxx17 ENABLE_DRAFTS
+)
+
```
Comment (JackBoosY) [inline ports/redis-plus-plus/portfile.cmake:15]:
```suggestion
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/redis-plus-plus/CONTROL [hunk 3]
```diff
@@ -1,5 +1,9 @@
...
 Version: 1.2.1
+Port-Version: 1
 Homepage: https://github.com/sewenew/redis-plus-plus
 Description: This is a C++ client for Redis. It's based on hiredis, and written in C++ 11.
-Build-Depends: hiredis
\ No newline at end of file
+Build-Depends: hiredis
+
+Feature: cxx17
+Description: Build redis-plus-plus using define flag REDIS_PLUS_PLUS_CXX_STANDARD=17
```
Comment (JackBoosY) [inline ports/redis-plus-plus/CONTROL:9]:
```suggestion
Description: Build redis-plus-plus with cxx 17 standard
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/redis-plus-plus/portfile.cmake [hunk 4]
```diff
@@ -9,6 +9,13 @@ vcpkg_from_github(
         fix-conversion.patch
 )
 
+if("cxx17" IN_LIST FEATURES)
+    set(REDIS_PLUS_PLUS_CXX_STANDARD 17)
+else()
+    set(REDIS_PLUS_PLUS_CXX_STANDARD 11)
+endif()
+
+
```
Comment (JackBoosY) [inline ports/redis-plus-plus/portfile.cmake:18]:
```suggestion
```

File: ports/redis-plus-plus/portfile.cmake [hunk 5]
```diff
@@ -20,11 +27,12 @@ endif()
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
-    OPTIONS
+    OPTIONS ${FEATURE_OPTIONS}
```
Comment (JackBoosY) [inline ports/redis-plus-plus/portfile.cmake:30]:
```suggestion
    OPTIONS
```

File: ports/redis-plus-plus/portfile.cmake [hunk 6]
```diff
@@ -34,4 +42,4 @@ vcpkg_copy_pdbs()
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 # Handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright )
\ No newline at end of file
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright )
```
Comment (JackBoosY) [inline ports/redis-plus-plus/portfile.cmake:45]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright )
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by strega-nil** [comment]
Thanks @JonLiu1993 :)

**Pull request merged**